core/templates/dev/head/exploration_editor/main_tab/StateEditor.js

```diff
@@ -49,23 +47,27 @@ oppia.controller('StateEditor', [
       $scope.initStateEditor();
     });
 
+    $scope.$on('refreshStateContent', function() {
+      $scope.content = explorationStatesService.getStateContentMemento(
+        editorContextService.getActiveStateName());
+    });
+
     $scope.$on('onInteractionIdChanged', function(evt, newInteractionId) {
       $scope.isInteractionIdSet = Boolean(newInteractionId);
       $scope.isCurrentStateTerminal = (
         $scope.isInteractionIdSet && INTERACTION_SPECS[
           newInteractionId].is_terminal);
     });
 
-    $scope.initStateEditor = function() {
-      $scope.stateName = editorContextService.getActiveStateName();
+    $scope.contentEditorIsOpen = false;
 
-      var stateData = explorationStatesService.getState($scope.stateName);
-      $scope.content = stateData.content;
-
-      // This should only be non-null when the content editor is open.
-      $scope.contentMemento = null;
+    $scope.initStateEditor = function() {
+      var stateName = editorContextService.getActiveStateName();
+      $scope.content = explorationStatesService.getStateContentMemento(
```

Comment:
Done -- moved everything into the 'if' block. I decided to use the `get*Memento()` methods instead, though -- better to move more towards that, I think. One day in the future we'll probably get rid of passing stateData all over the place.